src/decode/rr/draft_ietf_dnsop_svcb_https.rs

```diff
@@ -93,6 +44,67 @@ impl<'a, 'b: 'a> Decoder<'a, 'b> {
             https,
         })
     }
+
+    /// Decode a single service parameter
+    ///
+    /// Parameters:
+    /// - `service_parameter_key` - the IANA-controlled numeric identifier as defined in section
+    ///                             14.3 of the RFC
+    ///
+    /// Returns:
+    /// - `Ok(ServiceParameter)` - if there were no issues decoding the value
+    /// - `Err` - if there was any decoding error
+    fn rr_service_parameter(
+        &mut self,
+        service_parameter_key: u16,
```

Comment:
Could you move it to an extra `impl` block like the [`rr_edns_option_code`](https://github.com/LinkTed/dns-message-parser/blob/master/src/decode/rr/edns/rfc_6891.rs#L61) function because the self argument does not have the `'a` lifetime here?

Reply:
I moved this.